contribute/style-guides/e2e.md

```diff
@@ -0,0 +1,140 @@
+
+## Basic example
+Lets start with a simple example with a single selector and for simplicity all examples will be in JSX.
+
+In our example app we have an input that we during our e2e test want to type some text in.
+```jsx harmony
+<div>
+    <input type="text" className="gf-form-input login-form-input"/>
+</div>
+```
+
+We could define a selector using `JQuery` [type selectors](https://api.jquery.com/category/selectors/) with a string like `'.gf-form-input.login-form-input'` but that would be brittle as style changes occur frequently and there is nothing that signalizes to future developers that this input is part of an e2e test.
+
+At Grafana we've decided to use `aria-label` as our preferred way of defining our selectors instead of for instance`data-*` attributes because it also aids in accessibility.
+Lets add an descriptive `aria-label` to our simple example.
+```jsx harmony
+<div>
+    <input type="text" className="gf-form-input login-form-input" aria-label="Username input field"/>
+</div>
+```
+
+Now that we added the `aria-label` we suddenly get more information about this particular field. It's an input field that represents an username but there it's still not really signalizing that it's part of a e2e test.
+
+The next step is to create a `Page` representation in our e2e test framework to glue the test with the real implementation using the `pageFactory` function. For that function we can supply an `url` and `selectors` like the example below.
+```typescript
+export const Login = pageFactory({
+  url: '/login', // used when called from Login.visit()
+  selectors: {
+    username: 'Username input field', // used when called from Login.username().type('Hello World')
+  },
+});
+```
+
+The next step is to add `Login` page to the exported const `Pages` in `packages/grafana-e2e/src/pages/index.ts` so that it appears when we type `e2e.pages` in our IDE.
+```ecmascript 6
+export const Pages = {
+  Login,
+  ...,
+  ...,
+  ...,
+};
+
+```
+Now that we have a `Page` called `Login` in our `Pages` const we can use that to add a selector in our html like shown below and now this really signalizes to future developers that this is part of a e2e test.   
+```jsx harmony
+<div>
+    <input type="text" className="gf-form-input login-form-input" aria-label={e2e.pages.Login.selectors.username}/>
+</div>
+```
+
+The last step in our example is to use our `Login` page as part of a test. The `pageFactory` function we used before gives us 2 things by using it. 
+- Firstly the `url` property will be used whenever we call the `visit` function and is equivalent to the Cypress function [cy.visit()](https://docs.cypress.io/api/commands/visit.html#Syntax)
+- Secondly any defined selector in the `selectors` property can be accessed from `Login` page by invoking it and this is equivalent to the result of the Cypress function [cy.get(...)](https://docs.cypress.io/api/commands/get.html#Syntax) 
```

Comment:
```suggestion
- Secondly any defined selector in the `selectors` property can be accessed from the `Login` page by invoking it and this is equivalent to the result of the Cypress function [cy.get(...)](https://docs.cypress.io/api/commands/get.html#Syntax) 
```

Reply:
Remove "secondly" from the beginning of the sentence and add a period to the end.